src/Stateless/Transition.cs

```diff
@@ -45,6 +48,11 @@ public Transition(TState source, TState destination, TTrigger trigger)
             public TTrigger Trigger { get { return _trigger; } }
 
             /// <summary>
+            /// Optional data associated with the transition.
+            /// </summary>
+            public object Tag { get { return _tag; } }
```

Comment:
If you support guards, you need to be able to differentiate between transitions, and this is the only practical way we can come up with.
What I think is not necessary is the `Do()` feature, as that can be replaced with user specific implementation of the `Tag`, but I have no issue leaving it in as a helper feature.